### ISSUE TITLE
Add ability to adjust Ruler while the Magnifier updates in real time.

### ISSUE THREAD

**Issue by chipsnyder** [AvdLee/RocketSimApp#341]
**Is your feature request related to a problem? Please describe.**
When aligning two items with a verticle distance, I want to see their alignment using a ruler guideline and magnifier to see how far off they are from the ruler. 

To place the ruler, I'll:
- Place a ruler
- Magnify to see how close I am
- Repeat until I get the ruler to be the aligned where I want.
- Magnify the other location to check the alignment.

**Describe the solution you'd like**
I'd like to see the Magnify toggle on so I can shift the guideline to the spot that I want it.

**Describe alternatives you've considered**
Any solution that allows me to magnify and place a guideline would be awesome. I'd be happy with other solutions. 

**Additional context**
N/A

**Comment by chipsnyder** [comment]
Playing with your tool a bit more. If I can grab a guideline from anywhere on the line, or if it would only switch to the remove when dragging outside of the simulator then that would meet my needs too. Everything else works.

**Comment by AvdLee** [comment]
> If I can grab a guideline from anywhere on the line

Yeah, that would be nice, but I'm not aligned with how other apps handle rulers. On top of that: I want to prevent the rulers from taking over touches that should happen inside the Simulator app.

> or if it would only switch to the remove when dragging outside of the simulator then that would meet my needs too

"If it would only" makes it sound like it's now switching to remove inside of the simulator. That shouldn't be the case and sounds like a bug. Can you confirm?

----

I'm revisiting the Magnifier btw. I might want to add these options:
- Color picker #263 
- Keep magnifier enabled
- Disable magnifier to be bound to Simulator screen

I believe those should solve your request!

**Comment by chipsnyder** [comment]
Yeah, I think those would take care of my request especially the "Keep magnifier enabled."

> That shouldn't be the case and sounds like a bug. Can you confirm?

Here's a video of what I'm describing; it being a bug does make more sense. 

This video was recorded with RocketSim 10.0.6 but I just updated it to 10.2.1 and validated the same behavior there.
I'm on:
- M1 Macbook Pro Running Monterey 12.5 
- Xcode 14.1.0 

If you think it's an OS or Xcode version issue I can update it in a few weeks and re-test.

**Comment by AvdLee** [comment]
> or if it would only switch to the remove when dragging outside of the simulator then that would meet my needs too. Everything else works.

With your video this makes so much more sense now! This is definitely doable.
Btw, RocketSim 10.3.0 just became available with a few improvements to the magnifier for you to try out!

**Comment by chipsnyder** [comment]
Love the Magnifier always enabled 🙌 
The color identifier you added was also a major helper.
